bind.sh

```diff
@@ -133,15 +154,28 @@ fi
 if [[ -n "${cpus+x}" || -n "${mems+x}" ]]; then
   if command -v numactl &> /dev/null; then
       if [[ -n "${cpus+x}" ]]; then
-          set -- --physcpubind "${cpus[$rank]}" "$@"
+          set -- --physcpubind "${cpus[$local_rank]}" "$@"
       fi
       if [[ -n "${mems+x}" ]]; then
-          set -- --membind "${mems[$rank]}" "$@"
+          set -- --membind "${mems[$local_rank]}" "$@"
       fi
       set -- numactl "$@"
   else
       echo "Warning: numactl is not available, cannot bind to cores or memories" 1>&2
   fi
 fi
 
+# arguments may contain the substring %%LEGATE_GLOBAL_RANK%% which needs to be
+# be replaced with the actual computed rank for downstream processes to use
+count=0
+updated=()
+for arg in $@; do
+  updated+=("${arg/\%\%LEGATE_GLOBAL_RANK\%\%/$LEGATE_GLOBAL_RANK}")
+  count=$((count+1))
+done
+
+set -- "${updated[@]}"
+
+# echo "bind.sh: $@" 1>&2
```

Comment:
I guess printing this unconditionally is not good, but this is the ultimate command that actually get run, I do think we need some mechanism to surface this for debugging, etc.